src/runtime/audience-action-local-flow.ts

```diff
@@ -324,6 +330,37 @@ export class AudienceActionLocalFlow implements AudienceActionFlow {
     this.deps_.callbacks().triggerLoginRequest({linkRequested: false});
   }
 
+  private async complete_() {
+    const init: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+        'Accept': 'text/plain, application/json',
+      },
+      credentials: 'include',
+    };
+    const publicationId = this.deps_.pageConfig().getPublicationId();
+    const baseUrl = `/publication/${encodeURIComponent(
+      publicationId
+    )}/completeaudienceaction`;
+    const swgUserToken = await this.deps_
+      .storage()
+      .get(Constants.USER_TOKEN, true);
+    const queryParams = [
+      // TODO: mhkawano - check and error out if swgUserToken is null
+      ['sut', swgUserToken!],
+      // TODO: mhkawano - configurationId should not be optional
+      ['configurationId', this.params_.configurationId!],
+      ['audienceActionType', this.params_.action],
+    ];
+    const url = queryParams.reduce(
+      (url, [param, value]) => addQueryParam(url, param, value),
+      serviceUrl(baseUrl)
+    );
+    this.deps_.win().fetch(url, init);
+    // TODO: mhkawano - log error
```

Comment:
Maybe add a TODO to handle entitlement consumption logic on completion for paid content?

https://github.com/subscriptions-project/swg-js/blob/09df5f3068bcbff8d010eccfe7e679035fd09844/src/runtime/audience-action-flow.ts#L192

Reply:
Done.